### DISCUSSION TITLE
API to download or show file?

### DISCUSSION THREAD

**Discussion by wainggan** [rejetto/hfs#322] [Q&A]
Is there an API for plugins to download, delete, or etc, with a file?

Specifically anything that can replicate the functionality of these buttons, for example. 

![image](https://github.com/rejetto/hfs/assets/46691720/548b724b-43c1-4d2c-ae59-bd7934507e87)

Thank you for your time.

**Comment by rejetto** [comment]
yes!
all APIs for plugins are documented here https://github.com/rejetto/hfs/blob/main/dev-plugins.md
What you are looking for is the "fileMenu" event.
The "Enable tiles mode" is introduced by the "thumbnails" plugin, so you can also consider giving a look at it for an example. https://github.com/rejetto/thumbnails/blob/main/dist/public/main.js

**Comment by rejetto** [comment]
if you have a plugin idea in mind, consider sharing it and I may give hints on the easier way to get to it

**Reply by wainggan** [reply]
Hi!
What I'm imagining is a plugin that adds a simple button next to any file type that the "show" feature supports, which just opens that file in show. Exactly as it would if you clicked the file then selected show, but without the extra step.

**Comment by rejetto** [comment]
ok, so what you want to do is a "shortcut" to the file-show.
There's no api for that, currently.
Your best options is that you want for next release, so that I can give you access to the fileShow() function, so you'll be able to do `HFS.fileShow(file)`.
Otherwise you'd have to try to simulate the clicks, tricky.

On the web side, plugins are limited to the functions that I intentionally expose, while on the server side a plugin can access almost any function.

To add the button after the name you should use the event afterEntryName, that you can read about at
https://github.com/rejetto/hfs/blob/main/dev-plugins.md#front-end-api-events
You can see another plugin using it at https://github.com/IlyaBOT/mini-player/blob/main/dist/public/main.js
